proto/composerapis/astria/auctioneer/v1alpha1/bundle.proto

```diff
@@ -0,0 +1,39 @@
+syntax = "proto3";
+
+package astria.auctioneer.v1alpha1;
+
+import "google/protobuf/timestamp.proto";
+import "google/protobuf/empty.proto";
+import "astria/execution/v1alpha2/execution.proto";
+import "astria/auctioneer/v1alpha1/optimistic_block.proto";
+
+// Information for the bundle submitter to know how to submit the bundle.
+// The fee and base_sequencer_block_hash are not necessarily strictly necessary
+// it allows for the case where the server doesn't always send the highest fee
+// bundles after the previous but could just stream any confirmed bundles.
+message Bundle {
+    // The fee that can be expected to be received for submitting this bundle.
+    // This allows the bundle producer to stream any confirmed bundles they would be ok
+    // with submitting. Used to avoid race conditions in received bundle packets. Could
+    // also be used by a bundle submitter to allow multiple entities to submit bundles.
+    uint64 fee = 1;
```

Comment:
how is this verified? also what token is this in (rollup or sequencer)? i'm guessing rollup?

Reply:
this would be in rollup token and since we are running the auctioneer, we should be able to assume that this information passed between geth and the composer can be trusted, similar to how we can assume the information between a geth node and a conductor can be trusted.

Reply:
the fee will mainly be used to maintain the highest paying transaction seen so far in the auctioneer.

Reply:
another way would be for auctioneer to de-serialize the EVM transaction which would be more trustless but would be higher latency.

Reply:
> this would be in rollup token and since we are running the auctioneer, we should be able to assume that this information passed between geth and the composer can be trusted, similar to how we can assume the information between a geth node and a conductor can be trusted.

you mean passed between geth and auctioneer? so the fee is parsed by geth from the txs and verified in geth, then set to auctioneer essentially?

Reply:
yeah.

transactions are submitted to geth via the mempool rpc. when the geth mempool promotes transactions to pending they are checked for nonce and sufficient balance for the fee. geth provides a subscription for pending transactions (similar to what we do in composer) where promoted txs are sent to.
the geth-side bundle service will reap transactions from the pending transactions subscription (this is a `Subscription` object internal to geth that the websocket subscription also uses), and wrap them in this `Bundle` struct which it will then stream to the auctioneer.

because we plan to clear the mempool on geth after every optimistic block execution, the idea is that bundles received by the auctioneer via this stream after the optimistic execution result will have been checked against the optimistic state by geth's mempool's promotion to pending.

the auctioneer then has three streams, one for optimistic block execution results, one for block commitments and one for bundles. more on this in this part of the doc: https://www.notion.so/astria-org/20240919-Status-Update-1066bd31a90c80b68bb2ea9b898645d0?pvs=4#1066bd31a90c80f193e3d6cb72a2f6e9